flow/connectors/mongo/mongo.go

```diff
@@ -0,0 +1,431 @@
+		destinationTableName := req.TableNameMapping[sourceTableName].Name
+
+		items := model.NewRecordItems(2)
+
+		if documentKey, found := changeDoc["documentKey"]; found {
+			if len(documentKey.(bson.D)) == 0 || documentKey.(bson.D)[0].Key != DefaultDocumentKeyColumnName {
+				// should never happen
+				return errors.New("invalid document key, expect _id")
+			}
+			id := documentKey.(bson.D)[0].Value
+			qValue, err := qValueStringFromKey(id)
+			if err != nil {
+				return fmt.Errorf("failed to convert _id to string: %w", err)
+			}
+			items.AddColumn(DefaultDocumentKeyColumnName, qValue)
+		} else {
+			// should never happen
+			return errors.New("documentKey field not found")
+		}
+
+		if fullDocument, found := changeDoc["fullDocument"]; found {
+			qValue, err := qValueJSONFromDocument(fullDocument.(bson.D))
+			if err != nil {
+				return fmt.Errorf("failed to convert fullDocument to JSON: %w", err)
+			}
+			items.AddColumn(DefaultFullDocumentColumnName, qValue)
+		} else {
+			if changeDoc["operationType"] == "delete" {
+				items.AddColumn(DefaultFullDocumentColumnName, types.QValueJSON{Val: "{}"})
+			} else {
+				// should never happen with fullDocument='UpdateLookup'
+				return errors.New("fullDocument field not found")
+			}
+		}
+
+		if operationType, ok := changeDoc["operationType"]; ok {
+			switch operationType {
+			case "insert":
+				if err := addRecord(ctx, &model.InsertRecord[model.RecordItems]{
+					BaseRecord:           model.BaseRecord{CommitTimeNano: clusterTimeNanos},
+					Items:                items,
+					SourceTableName:      sourceTableName,
+					DestinationTableName: destinationTableName,
+				}); err != nil {
+					return fmt.Errorf("failed to add insert record: %w", err)
+				}
+			case "update", "replace":
+				if err := addRecord(ctx, &model.UpdateRecord[model.RecordItems]{
+					BaseRecord:           model.BaseRecord{CommitTimeNano: clusterTimeNanos},
+					NewItems:             items,
+					SourceTableName:      sourceTableName,
+					DestinationTableName: destinationTableName,
+				}); err != nil {
+					return fmt.Errorf("failed to add update record: %w", err)
+				}
+			case "delete":
+				if err := addRecord(ctx, &model.DeleteRecord[model.RecordItems]{
+					BaseRecord:           model.BaseRecord{CommitTimeNano: clusterTimeNanos},
+					Items:                items,
+					SourceTableName:      sourceTableName,
+					DestinationTableName: destinationTableName,
+				}); err != nil {
+					return fmt.Errorf("failed to add delete record: %w", err)
+				}
+			default:
+				return fmt.Errorf("unsupported operationType: %s", operationType)
+			}
+		}
+	}
+	if resumeToken := changeStream.ResumeToken(); resumeToken != nil {
+		// Update the last offset with the resume token
+		req.RecordStream.UpdateLatestCheckpointText(base64.StdEncoding.EncodeToString(resumeToken))
+		c.logger.Info("Updated last offset to: " + req.LastOffset.Text)
```

Comment:
```suggestion
		c.logger.Info("[mongo] latest resume token", slog.String("resumeToken", req.LastOffset.Text))
```
"updated" is a bit misleading: that should be reserved for when this is persisted to catalog